iohk/iohk-ops.hs

```diff
@@ -11,20 +11,28 @@ import qualified Filesystem.Path.CurrentOS     as Path
 import           Text.Read                        (readMaybe)
 import           Turtle                    hiding (procs, shells)
 
-import           NixOps                           (Branch(..), Commit(..), Environment(..), Deployment(..), Target(..)
+import           NixOps                           (Branch(..), Commit(..), Environment(..), Deployment(..), Target(..), NodeName(..)
                                                   ,Options(..), NixopsCmd(..), Project(..), Region(..), URL(..)
-                                                  ,showT, cmd, incmd, projectURL)
+                                                  ,showT, lowerShowT, errorT, cmd, incmd, projectURL)
 import qualified NixOps                        as Ops
 import qualified CardanoCSL                    as Cardano
 import qualified Timewarp                      as Timewarp
 
 
 -- * Elementary parsers
 --
-optReadLower :: Read a => ArgName -> ShortName -> Optional HelpMessage -> Parser a
-optReadLower = opt (readMaybe . T.unpack . T.toTitle)
-argReadLower :: Read a => ArgName -> Optional HelpMessage -> Parser a
-argReadLower = arg (readMaybe . T.unpack . T.toTitle)
+diagReadMaybe :: (Bounded a, Enum a, Read a, Show a) => String -> Maybe a
```

Comment:
This seems fairly complicated, why don't we use our own `Read` instance instead?